### PR TITLE
Update Types

### PR DESCRIPTION
**Pull request by adrice727**
- Remove `webpack` and use `TS` compiler to build project
  -> Add `es2015.promise` to `tsconfig` and target `ES5`
- Remove `tsdocs` from the build
- Refactor types across project
  -> New namespace for `OT`
  -> Replace global types (i.e. `.d.ts` files) with local types (modules)
  -> Use latest version of `opentok-solutions-logging`
- Include type declarations in `npm` package

### CODE REVIEW

**Comment by jeffswartz** [comment]
@adrice727 could you redo this PR and point it to the develop branch?

File: src/NetworkTest/testConnectivity/index.ts [hunk 1]
```diff
@@ -10,10 +10,17 @@
  */
 import axios from 'axios';
 import * as Promise from 'promise';
+/* tslint:disable */
```
[the remark this answers is not in the document]
Comment (adrice727) [reply src/NetworkTest/testConnectivity/index.ts]: `tslint` will throw a misnamed import error.

File: src/NetworkTest/types/opentok/index.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import * as OTSubscriber from './subscriber';
+import * as OTError from './error';
+import * as OTEvent from './events';
+
+export namespace OT {
+
+  /**
+   * OpenTok.js Client SDK
+   */
+  export interface Client {
```
[the remark this answers is not in the document]
Comment (adrice727) [reply src/NetworkTest/types/opentok/index.ts]: We can't have abstract functions in a namespace outside of a `d.ts` file.  I'd have to think of another way to do it.  I don't think it matters too much for the time being.  We can replace all of this with the types from the `@opentok/client` package as soon as the types are updated.

File: package.json [hunk 3]
```diff
@@ -60,14 +56,8 @@
     "karma-sourcemap-loader": "^0.3.7",
     "karma-webpack": "^2.0.13",
     "opentok": "^2.6.2",
-    "string-replace-loader": "^1.3.0",
-    "ts-loader": "^3.5.0",
```
[the remark this answers is not in the document]
Comment (adrice727) [reply package.json]: `karma.conf.js` is now using `awesome-typescript-loader`.

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,5109 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
```
[the remark this answers is not in the document]
Comment (adrice727) [reply yarn.lock]: I removed `yarn-lock` and added it to `.gitignore`.

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jeffswartz) [reply package.json]: OK. I no longer get the error.

File: src/NetworkTest/testConnectivity/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeffswartz) [reply src/NetworkTest/testConnectivity/index.ts]: Right. I see that now.

File: src/NetworkTest/types/opentok/index.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeffswartz) [reply src/NetworkTest/types/opentok/index.ts]: OK.

**Pull request merged**